packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -161,6 +192,24 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
         };
         updateFocusContextKeys();
         this.shell.activeChanged.connect(updateFocusContextKeys);
+        const widget = await this.widget;
+        const model: FileNavigatorModel = widget.model;
```

Comment:
caching widget and model in the constructor is a memory leak, please always get them in `this.onDidCreateNewResource`.

Don't use `this.widget` it will create a new instance even it is not opened, use `this.tryGetWidget()` manybe also add a check `widget.isVisible` to make sure that there was a navigator opened and revealed to a user, otherwise nothing should happen.

We should check a case: close the navitaor, use file menu to create a new file, navigator should not be revealed.

Reply:
Done, tested with closed navigator.